### PR TITLE
/go/store/prolly: flatbuffers-based prolly tree

### PR DESCRIPTION
**Pull request by andrew-wm-arthur**
- replaces `Node` with flatbuffers-based `mapNode`
- temporarily removes map-count accounting from `Map`

### CODE REVIEW

File: go/store/prolly/node_store.go [hunk 1]
```diff
@@ -32,12 +32,11 @@ const (
 
 // NodeStore reads and writes prolly tree Nodes.
 type NodeStore interface {
+	// Read reads a prolly tree mapNode from the store.
+	Read(ctx context.Context, ref hash.Hash) (mapNode, error)
```
Comment (reltuk) [inline go/store/prolly/node_store.go:36]: How do people outside this package using this public interface use this private type (mapNode)?

**Review by reltuk — APPROVED** [review]
Seems great to me. A few comments.

File: go/store/prolly/mutable_map_write_test.go [hunk 2]
```diff
@@ -89,7 +89,7 @@ func testPointUpdates(t *testing.T, mapCount int) {
 		require.NoError(t, err)
 
 		m := materializeMap(t, mut)
-		assert.Equal(t, mapCount, int(m.Count()))
+		//assert.Equal(t, mapCount, int(m.Count()))
```
[the remark this answers is not in the document]
Comment (andrew-wm-arthur) [reply go/store/prolly/mutable_map_write_test.go:92]: I'm going to add back cumulative count eventually. Will update tests for now

File: go/store/prolly/node.go [hunk 3]
```diff
@@ -17,204 +17,186 @@ package prolly
...
-func (p nodePair) key() nodeItem {
-	return p[0]
-}
+func makeMapNode(pool pool.BuffPool, level uint64, keys, values []nodeItem) (node mapNode) {
+	var keySz, valSz int
+	for i := range keys {
+		keySz += len(keys[i])
+		valSz += len(values[i])
+	}
+	b := getMapBuilder(pool, keySz+valSz+96)
```
[the remark this answers is not in the document]
Comment (andrew-wm-arthur) [reply go/store/prolly/node.go:49]: rough estimate of node metadata + vtable size

File: go/store/prolly/node_store.go (hunk 1 above)
Comment (andrew-wm-arthur) [reply go/store/prolly/node_store.go:36]: couldn't remember why `Node` was public, this is it

**Pull request merged**